src/components/common/input.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Input as AntInput, Form } from 'antd';
+import { EyeInvisibleOutlined, EyeTwoTone } from '@ant-design/icons';
+
+interface InputProps {
+  placeholder?: string;
+  type?: 'text' | 'password' | 'email';
+  name: string;
+  [key: string]: unknown;
+}
+
+const Input: React.FC<InputProps> = ({ placeholder, type = 'text', name, ...props }) => {
+  const commonClasses = 'bg-[#EAF0F7] border-none p-2 color-[#4F555A]';
+
+  const emailValidator = (_rule: unknown, value: string) => {
+    if (!value) {
+      return Promise.reject('Please input your email!');
+    }
+    const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+    if (!emailRegex.test(value)) {
+      return Promise.reject('Please input a valid email!');
+    }
+    return Promise.resolve();
+  };
+
+  const passwordValidator = (_rule: unknown, value: string) => {
+    if (!value) {
+      return Promise.reject('Please input your password!');
+    }
+    const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+    if (!passwordRegex.test(value)) {
+      return Promise.reject(
+        'Password must be at least 8 characters long and contain both letters and numbers!'
+      );
+    }
+    return Promise.resolve();
+  };
+
+  return (
+    <Form.Item
```

Comment:
<Form.Item
        label="Password"
        name="password"
        rules={passwordRules}
      >
      
      just using when you call.
      Don't using ChatGPT too much, it made you can't thing everything